### PR TITLE
IBX-6043: Owner criterion added in Shipment and Order search criteria

### PR DESCRIPTION
**Pull request by julitafalcondusza**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | (https://issues.ibexa.co/browse/IBX-5994)
| Versions      | master
| Edition       | Commerce

Owner criterion added in Shipment and Order search criteria

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Code samples are working
- [x] PHP code samples have been fixed with PHP CS fixer
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/search/criteria_reference/order_owner_criterion.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+description: Owner Criterion
+edition: commerce
+---
+
+# Owner Criterion
+
+The `Owner` Criterion searches for orders based on the user reference.
```
Comment (konradoboza) [inline docs/search/criteria_reference/order_owner_criterion.md:8]: Name is slightly different than in Shipments/Payments:
```suggestion
The `OwnerCriterion` Criterion searches for orders based on the user reference.
```

File: docs/search/criteria_reference/order_owner_criterion.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Arguments
+
+- `user reference` - \Ibexa\Contracts\Core\Repository\Values\User\UserReference(int $userId)
```
Comment (konradoboza) [inline docs/search/criteria_reference/order_owner_criterion.md:12]:
```suggestion
- `UserReference` object - \Ibexa\Contracts\Core\Repository\Values\User\UserReference(int $userId)
```

File: docs/search/criteria_reference/order_owner_criterion.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+``` php
+$query = new OrderQuery(
+    new \Ibexa\Contracts\OrderManagement\Value\Order\Query\Criterion\OwnerCriterion(
+        \Ibexa\Contracts\Core\Repository\Values\User\UserReference(14)
+    )
+);
+```
+
+Owner Criterion accepts also owners board:
```
Comment (konradoboza) [inline docs/search/criteria_reference/order_owner_criterion.md:24]:
```suggestion
`OwnerCriterion` accepts also multiple values:
```

File: docs/search/criteria_reference/order_search_criteria.md [hunk 4]
```diff
@@ -19,6 +19,7 @@ With these Criteria you can filter orders by their order identifier, order creat
 |[CustomerNameCriterion](order_customer_name_criterion.md)|Customer's user name|
 |[IdentifierCriterion](order_identifier_criterion.md)|Order identifier|
 |[IsCompanyAssociatedCriterion](order_company_associated_criterion.md)|Whether the customer represents a company|
+|[OwnerCriterion](order_owner_criterion.md)|Owner identifier based on the user reference|
```
Comment (konradoboza) [inline docs/search/criteria_reference/order_search_criteria.md:22]:
```suggestion
|[OwnerCriterion](order_owner_criterion.md)|Owner based on the user reference|
```

File: docs/search/criteria_reference/shipment_owner_criterion.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+edition: commerce
+---
+
+# Owner Criterion
+
+The `Owner` Criterion searches for shipments based on the user reference.
+
+## Arguments
+
+- `user reference` - \Ibexa\Contracts\Core\Repository\Values\User\UserReference(int $userId)
```
Comment (konradoboza) [inline docs/search/criteria_reference/shipment_owner_criterion.md:12]: Same as above.

File: docs/search/criteria_reference/shipment_owner_criterion.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+``` php
+$query = new ShipmentQuery(
+    new \Ibexa\Contracts\Shipping\Shipment\Query\Criterion\Owner(
+        \Ibexa\Contracts\Core\Repository\Values\User\UserReference(14)
+    )
+);
+```
+
+Owner Criterion accepts also owners board:
```
Comment (konradoboza) [inline docs/search/criteria_reference/shipment_owner_criterion.md:24]:
```suggestion
Owner Criterion accepts also multiple values:
```

File: docs/search/criteria_reference/shipment_search_criteria.md [hunk 7]
```diff
@@ -19,6 +19,7 @@ With these Criteria you can filter shipments by their shipment identifier, shipm
 |[Identifier](shipment_identifier_criterion.md)|Shipment identifier|
 |[LogicalAnd](shipment_logicaland_criterion.md)|Logical AND criterion that matches if all the provided Criteria match|
 |[LogicalOr](shipment_logicalor_criterion.md)|Logical OR criterion that matches if at least one of the provided Criteria matches|
+|[Owner](shipment_owner_criterion.md)|Owner identifier based on the user reference|
```
Comment (konradoboza) [inline docs/search/criteria_reference/shipment_search_criteria.md:22]:
```suggestion
|[Owner](shipment_owner_criterion.md)|Owner based on the user reference|
```

**Review by konradoboza — CHANGES_REQUESTED** [review]

**Review by konradoboza — APPROVED** [review]

**Review by radzyu — APPROVED** [review]

**Review by adamwojs — APPROVED** [review]

**Review by juskora — APPROVED** [review]

**Pull request merged**